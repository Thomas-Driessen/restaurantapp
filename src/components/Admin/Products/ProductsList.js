import React from 'react';
import Grid from '@material-ui/core/Grid';
import Product from './Product';
import Typography from "@material-ui/core/Typography";

const ProductsList = (props) => {
    return(
        <Grid container spacing={1} style={{padding: 10, paddingTop: 10}}>
            { props.categories.map(currentCategory => (
                <div style={{width: "100%", padding: 10, paddingTop: 10}}>
                    <Typography variant="inherit" component="h2">
                        {currentCategory.categoryName}
                    </Typography>
                    <Grid key={currentCategory.id} container spacing={1}>
                        {currentCategory.products.map(currentProduct => (
                            <Grid item xs={12} sm={6} lg={4} xl={3} key={currentProduct.id} style={{padding: 10}}>
                                <Product key={currentProduct.id}
                                         product={currentProduct}
                                         category={currentCategory}
                                         foodCategories={props.foodCategories}
                                         drinkCategories={props.drinkCategories}
                                         ingredients={props.ingredients}
                                         productType={props.productType} />
                            </Grid>
                        ))}
                    </Grid>
                </div>
            ))}
        </Grid>
    )
}
export default ProductsList