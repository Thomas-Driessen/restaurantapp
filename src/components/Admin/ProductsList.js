import React from 'react';
import Grid from '@material-ui/core/Grid';
import Product from './Product';

const ProductsList = (props) => {
    return(
    <Grid container spacing={1} style={{padding: 10}}>
        { props.products.map((currentProduct, index) => (
            <Grid key={index} item xs={12} sm={6} lg={4} xl={3}>
                <Product key={index} product={currentProduct} productType={props.productType}/>
            </Grid>
        ))}
    </Grid>
    )
}
export default ProductsList