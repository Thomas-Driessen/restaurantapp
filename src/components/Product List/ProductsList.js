import React from 'react';
import Grid from '@material-ui/core/Grid';
import Product from './Product';

const ProductsList = (props) => {
    return (
        <Grid container spacing={2} style={{ padding: 20 }}>
            { props.products.map(currentProduct => (
                <Grid key={currentProduct.id} item sm={12} lg={6}>
                    <Product key={currentProduct.id} product={currentProduct} productType={props.productType} />
                </Grid>
            ))}
        </Grid>
    )
}
export default ProductsList