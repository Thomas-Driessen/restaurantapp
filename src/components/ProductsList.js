import React from 'react';
import Grid from '@material-ui/core/Grid';
import Product from './Product';
import productData from '../products';
const ProductsList = () => {
    return(
    <Grid container spacing={24} style={{padding: 24}}>
        { productData.map(currentProduct => (
            <Grid item xs={12} sm={6} lg={4} xl={3}>
                <Product product={currentProduct} />
            </Grid>
        ))}
    </Grid>
    )
}
export default ProductsList