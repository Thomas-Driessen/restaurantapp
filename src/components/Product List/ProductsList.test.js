import React from 'react';
import { render } from '@testing-library/react';
import ProductsList from './ProductsList';
import renderer from 'react-test-renderer';

const products = [
  {
    id: 1,
    title: "Coca-Cola",
    price: 2.5,
    description: ""
  },
  {
    id: 2,
    title: "Pepsi",
    price: 2,
    description: ""
  },
];

test('Render Product Coca-Cola', () => {
  const { getByText } = render(<ProductsList products={products} />);
  const linkElement = getByText(/Coca-cola/i);
  expect(linkElement).toBeInTheDocument();
});

test('Render Product Pepsi', () => {
  const { getByText } = render(<ProductsList products={products} />);
  const linkElement = getByText(/Pepsi/i);
  expect(linkElement).toBeInTheDocument();
});

it('ProductsList renders correctly', () => {
  const tree = renderer
    .create(<ProductsList products={products} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

afterAll(() => {
  global.gc && global.gc()
})