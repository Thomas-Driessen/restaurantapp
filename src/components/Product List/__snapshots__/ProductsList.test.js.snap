// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  style={
    Object {
      "padding": 15,
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 MuiGrid-grid-xl-3"
  >
    <div>
      <div>
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt="Image for Coca-Cola Not Found"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="250"
            src="../../images/undefined/1.jpg"
            style={
              Object {
                "height": 400,
              }
            }
            title="Coca-Cola"
          />
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-gutterBottom"
            >
              Coca-Cola
            </h2>
            <h6
              className="MuiTypography-root MuiTypography-body1"
            />
            <p
              className="MuiTypography-root makeStyles-price-3 MuiTypography-body1"
            >
              2.5
              €
            </p>
          </div>
          <div
            className="MuiCardActions-root MuiCardActions-spacing"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-flex-end"
            >
              <div
                style={
                  Object {
                    "alignItems": "right",
                    "display": "flex",
                  }
                }
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    View Details
                  </span>
                </button>
                <button
                  aria-label="add to favorites"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  aria-label="add to order"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 MuiGrid-grid-xl-3"
  >
    <div>
      <div>
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt="Image for Pepsi Not Found"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="250"
            src="../../images/undefined/2.jpg"
            style={
              Object {
                "height": 400,
              }
            }
            title="Pepsi"
          />
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-gutterBottom"
            >
              Pepsi
            </h2>
            <h6
              className="MuiTypography-root MuiTypography-body1"
            />
            <p
              className="MuiTypography-root makeStyles-price-3 MuiTypography-body1"
            >
              2
              €
            </p>
          </div>
          <div
            className="MuiCardActions-root MuiCardActions-spacing"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-flex-end"
            >
              <div
                style={
                  Object {
                    "alignItems": "right",
                    "display": "flex",
                  }
                }
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    View Details
                  </span>
                </button>
                <button
                  aria-label="add to favorites"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  aria-label="add to order"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
